src/coreclr/inc/bundle.h

```diff
@@ -43,8 +43,9 @@ class Bundle
     Bundle(LPCSTR bundlePath, BundleProbeFn *probe);
     BundleFileLocation Probe(const SString& path, bool pathIsBundleRelative = false) const;
 
-    const SString &Path() const { LIMITED_METHOD_CONTRACT; return m_path; }
-    const SString &BasePath() const { LIMITED_METHOD_CONTRACT; return m_basePath; }
+    const SString& Path() const { LIMITED_METHOD_CONTRACT; return m_path; }
```

Comment:
I see we're following the pattern here so push back if the risks are mitigated. Would it make more sense to return `WCHAR*` or `LPUTF8*` and avoid exposing the underlying `SString`? In the few places I noted there is already a copy. Perhaps preemptively making a useable copy is safer?

Reply:
There's no real mitigation other than that we don't do anything to modify it / don't expect any usage to should. I get the desire, but I'm kind of ambivalent about making the consumption a bit less consistent and nice to use because the `const`-ness is a lie.

Reply:
Switch `ExtractionPath` and `BasePath` to `WCHAR*`/`UTF8*`. Left `Path` alone, as I think that would be a bit more involved.